api/src/api/schemas/search_schema.py

```diff
@@ -102,6 +122,14 @@ def with_one_of(
         self.schema_fields["one_of"] = fields.List(list_type, validate=[validators.Length(min=1)])
 
         return self
+    
+    def with_start_date(self) -> "StrSearchSchemaBuilder":
+        self.schema_fields["start_date"] = fields.Date(allow_none=True)
+        return self
```

Comment:
Instead of re-using the StrSearchSchemaBuilder, could you create a new date builder?

Might be worth making a base class that has the init/build methods shared across them.

Reply:
I think I have accomplished this with the latest commit. Please let me know what you think.